iothub/device/src/Transport/AmqpIoT/AmqpIoTResourceException.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Microsoft.Azure.Devices.Client.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Azure.Devices.Client.Transport.AmqpIoT
+{
+    internal class AmqpIoTResourceException : IotHubException
```

Comment:
This is thrown by the API. 
We should never throw internal types as it makes it extremely hard to perform operations and identify what happened.

Reply:
External user should perform operations as an `IotHubException`.

Reply:
Discussion on internal exceptions thrown publicly. Table P0